app/static/js/index.js

```diff
@@ -356,6 +377,20 @@ function loadNextPage() {
     }
 }
 
+function updateCriteria(criteria) {
+    let urlSearchParams = new URLSearchParams(window.location.search);
+    let params = Object.fromEntries(urlSearchParams.entries());
+    if (params['query'] != 'undefined') {
```

Comment:
It would be preferable, if possible, to never even add a query parameter to the URL if there's nothing in it